storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/snowflake/SnowflakeTemplater.java

```diff
@@ -47,79 +54,133 @@ public String getTableCreationSql(FeatureSetProto.FeatureSetSpec featureSetSpec)
-        //            String.format(
-        //                "Found column %s in table %s that should not exist", existingColumn,
-        // tableName));
-        tableMigrationSql =
-            String.format("ALTER TABLE %s DROP COLUMN %s;", tableName, existingColumn);
-        tableMigrationSql = tableMigrationSql + "\n";
-        System.out.println("tableMigrationSql-----" + tableMigrationSql);
-      }
-      requiredColumns.remove(existingColumn.toLowerCase());
-    }
+  public String getFeatureRowInsertSql(FeatureRowProto.FeatureRow element, String jobName) {
+
+    StringJoiner columnsSql = new StringJoiner(",");
+    StringJoiner valueSql = new StringJoiner(",");
 
-    if (requiredColumns.size() == 0) {
-      log.info(
-          String.format("All columns already exist for table %s, no update necessary.", tableName));
-      return tableMigrationSql;
+    String tableName = JdbcTemplater.getTableNameFromFeatureSet(element.getFeatureSet());
+    //    Map<String, String> requiredColumns = getRequiredColumns(featureSetSpec);
+    Map<String, String> requiredColumns = getRequiredColumns();
+    for (String column : requiredColumns.keySet()) {
+      columnsSql.add(column);
     }
+    Map<String, ValueProto.Value> fieldMap =
+        element.getFieldsList().stream()
+            .collect(Collectors.toMap(FieldProto.Field::getName, FieldProto.Field::getValue));
+
+    // Set event_timestamp
+    Instant eventTsInstant =
+        Instant.ofEpochSecond(element.getEventTimestamp().getSeconds())
+            .plusNanos(element.getEventTimestamp().getNanos());
 
-    StringJoiner addColumnSql = new StringJoiner(", ");
-    // Filter required columns down to only the ones we need to add
-    for (String requiredColumn : requiredColumns.keySet()) {
-      String requiredColumnType = requiredColumns.get(requiredColumn);
-      addColumnSql.add(String.format("ADD COLUMN %s %s", requiredColumn, requiredColumnType));
+    valueSql.add("'" + Timestamp.from(eventTsInstant) + "'");
+
+    // Set created_timestamp
+
+    valueSql.add("'" + System.currentTimeMillis() + "'");
+
+    // Set Feature
+
+    JSONObject json_variant = new JSONObject();
+    for (String row : fieldMap.keySet()) {
+      setFeatureValue(json_variant, row, fieldMap.get(row));
     }
 
-    tableMigrationSql =
-        tableMigrationSql + String.format("ALTER TABLE %s %s;", tableName, addColumnSql);
-    log.debug(tableMigrationSql);
-    return tableMigrationSql;
-  }
+    valueSql.add("parse_json('" + json_variant.toString() + "')");
 
-  public String getFeatureRowInsertSql(FeatureSetProto.FeatureSetSpec featureSetSpec) {
-    StringJoiner columnsSql = new StringJoiner(",");
-    StringJoiner valueSql = new StringJoiner(",");
+    // Set ingestion Id
 
-    Map<String, String> requiredColumns = getRequiredColumns(featureSetSpec);
+    if (element.getIngestionId() == "") {
+      valueSql.add("NULL");
+    } else {
+      valueSql.add("'" + element.getIngestionId() + "'");
```

Comment:
use parameter binding not string concatenation

Reply:
@dr3s did you mean preparedStatement.setString() method?

Reply:
yes